includes/Admin/Admin_AJAX.php

```diff
@@ -46,6 +50,100 @@ public function get_nonce() {
 		return wp_create_nonce( self::NONCE_KEY );
 	}
 
+	/**
+	 * Handles the AJAX request to setup the runtime environment if needed.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function setup_environment() {
+		// Verify the nonce before continuing.
+		$valid_nonce = $this->verify_nonce( filter_input( INPUT_POST, 'nonce', FILTER_SANITIZE_STRING ) );
+
+		if ( is_wp_error( $valid_nonce ) ) {
+			wp_send_json_error( $valid_nonce, 403 );
+		}
+		$runner = Plugin_Request_Utility::get_runner();
+
+		if ( is_null( $runner ) ) {
+			$runner = new AJAX_Runner();
+		}
+
+		// Make sure we are using the correct runner instance.
+		if ( ! ( $runner instanceof AJAX_Runner ) ) {
+			wp_send_json_error(
+				new WP_Error( 'invalid-runner', __( 'AJAX Runner was not initialized correctly.', 'plugin-check' ) ),
+				500
+			);
+		}
+
+		$checks = wp_parse_list( filter_input( INPUT_POST, 'checks', FILTER_SANITIZE_STRING ) );
+		$plugin = filter_input( INPUT_POST, 'plugin', FILTER_SANITIZE_STRING );
+
+		try {
+			$runner->set_check_slugs( $checks );
+			$runner->set_plugin( $plugin );
+			$checks_to_run = $runner->get_checks_to_run();
+		} catch ( Exception $error ) {
+			wp_send_json_error(
+				new WP_Error( 'invalid-request', $error->getMessage() ),
+				400
+			);
+		}
+
+		$message = __( 'No runtime checks, runtime environment was not setup.', 'plugin-check' );
+
+		if ( $this->has_runtime_check( $checks_to_run ) ) {
+			$runtime = new Runtime_Environment_Setup();
+			$runtime->setup();
+			$message = __( 'Runtime environment setup successful.', 'plugin-check' );
+		}
+
+		wp_send_json_success(
+			array(
+				'message' => $message,
+				'plugin'  => $plugin,
+				'checks'  => $checks,
+			)
+		);
+	}
+
+	/**
+	 * Handles the AJAX request to cleanup the runtime environment.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function cleanup_environment() {
+		global $wpdb;
+
+		// Verify the nonce before continuing.
+		$valid_nonce = $this->verify_nonce( filter_input( INPUT_POST, 'nonce', FILTER_SANITIZE_STRING ) );
+
+		if ( is_wp_error( $valid_nonce ) ) {
+			wp_send_json_error( $valid_nonce, 403 );
+		}
+
+		// Set the new prefix.
+		$old_prefix = $wpdb->set_prefix( 'wppc_' );
+
+		$message = __( 'Runtime environment was not prepared, cleanup was not run.', 'plugin-check' );
+
+		// Test if the runtime environment tables exist.
+		if ( 'wppc_posts' === $wpdb->get_var( "SHOW TABLES LIKE 'wppc_posts'" ) ) {
```

Comment:
Maybe we should also check for the constant that our `object-cache.php` drop-in sets. Just to make sure that we also run the clean up if e.g. the `object-cache.php` drop-in was placed but for some reason the database tables were not created successfully.
```suggestion
		if ( 'wppc_posts' === $wpdb->get_var( "SHOW TABLES LIKE 'wppc_posts'" ) || defined( 'WP_PLUGIN_CHECK_OBJECT_CACHE_DROPIN_VERSION' ) ) {
```

Reply:
@felixarntz I agree and have updated